base/file.jl

```diff
@@ -416,18 +416,29 @@ function touch(path::AbstractString)
     path
 end
 
-const temp_prefix = "jl_"
-
-if Sys.iswindows()
+"""
+    tempdir()
 
+Obtain the path of a temporary directory (possibly shared with other processes). On Windows,
+`tempdir()` uses the first environment variable found in the ordered list `TMP`, `TEMP`,
+`USERPROFILE`. On all other operating systems, `tempdir()` uses the first environment
+variable found in the ordered list `TMPDIR`, `TMP`, `TEMP`, and `TEMPDIR`. If none of these are
+found, the path `"/tmp"` is used.
+"""
 function tempdir()
-    temppath = Vector{UInt16}(undef, 32767)
-    lentemppath = ccall(:GetTempPathW, stdcall, UInt32, (UInt32, Ptr{UInt16}), length(temppath), temppath)
-    windowserror("GetTempPath", lentemppath >= length(temppath) || lentemppath == 0)
-    resize!(temppath, lentemppath)
-    return transcode(String, temppath)
+    path_max = 1024
+    buf = Base.StringVector(path_max)
+    sz = Ref{Csize_t}(path_max + 1)
+    rc = ccall(:uv_os_tmpdir, Cint, (Ptr{UInt8}, Ptr{Csize_t}), buf, sz)
+    rc != 0 && error("unable to retrieve temporary directory")
```

Comment:
Use e.g. `uv_error("tempdir", rc)`, which checks the error code and gives an appropriate message.